ERCS/erc-7683.md

```diff
@@ -98,76 +111,174 @@ struct Input {
+	/// @dev To be called by the user
+	/// @dev This method must emit the Initiated event
+	/// @param order The OnchainCrossChainOrder definition
+	/// @param signature The user's signature over the order
 	/// @param fillerData Any filler-defined data required by the settler
-	function initiate(CrossChainOrder order, bytes signature, bytes fillerData) external;
+	/// @returns fillInstructions instructions to parameterize all fill legs required by this order
+	function initiate(OnchainCrossChainOrder order) external returns (FillInstructions[] fillInstructions);
+
+	/// @notice Resolves a specific GaslessCrossChainOrder into a generic ResolvedCrossChainOrder
+	/// @dev Intended to improve standardized integration of various order types and settlement contracts
+	/// @param order The GaslessCrossChainOrder definition
+	/// @param originFillerData Any filler-defined data required by the settler
+	/// @returns ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+	function resolveFor(GaslessCrossChainOrder order, bytes originFillerData) external view returns (ResolvedCrossChainOrder);
 
-	/// @notice Resolves a specific CrossChainOrder into a generic ResolvedCrossChainOrder
+	/// @notice Resolves a specific OnchainCrossChainOrder into a generic ResolvedCrossChainOrder
 	/// @dev Intended to improve standardized integration of various order types and settlement contracts
-	/// @param order The CrossChainOrder definition
+	/// @param order The OnchainCrossChainOrder definition
 	/// @param fillerData Any filler-defined data required by the settler
 	/// @returns ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
-	function resolve(CrossChainOrder order, bytes fillerData) external view returns (ResolvedCrossChainOrder);
+	function resolve(OnchainCrossChainOrder order) external view returns (ResolvedCrossChainOrder);
+}
+```
+
+A compliant destination settlement contract implementation MUST implement the `IDestinationSettler` interface:
+
+```solidity
+/// @title IDestinationSettler
+/// @notice Standard interface for settlement contracts on the destination chain
+interface IDestinationSettler {
+	/// @notice Fills a single leg of a particular order on the destination chain
+	/// @param orderId Unique order identifier for this order
+	/// @param originData Data emitted on the origin to parameterize the fill
+	/// @param fillerData Data provided by the filler to inform the fill or express their preferences
+	function fill(bytes32 orderId, bytes calldata originData, bytes calldata fillerData) external;
 }
 ```
 
+Cross-chain execution systems implementing this standard SHOULD use a sub-type that can be parsed from the arbitrary `fillerData` field. This may include information such as the desired timing or form of payment for the filler
+
+All sub-types SHOULD be registered at github.com/erc-7683/filler-data-subtypes to encourage sharing of sub-types based on their functionality.
```

Comment:
I wonder if there will be overlap between filler and order subtypes and how we can avoid requiring dApps to publish duplicate instructions in the 7683 repo

Reply:
I wasn't thinking there would be overlaps, since I think they serve pretty different purposes. Do you have any examples of where you think they might overlap?

Reply:
hmm will think, good push